app/org/sagebionetworks/bridge/util/BulkDownloadUtil.java

```diff
@@ -0,0 +1,249 @@
+package org.sagebionetworks.bridge.util;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperConfig;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.google.common.base.Charsets;
+import com.google.common.base.Joiner;
+import com.google.common.io.Files;
+import org.joda.time.format.ISODateTimeFormat;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+import org.sagebionetworks.bridge.dynamodb.DynamoHealthCode;
+import org.sagebionetworks.bridge.dynamodb.DynamoStudy;
+import org.sagebionetworks.bridge.dynamodb.DynamoUpload2;
+import org.sagebionetworks.bridge.upload.DecryptHandler;
+import org.sagebionetworks.bridge.upload.ParseJsonHandler;
+import org.sagebionetworks.bridge.upload.S3DownloadHandler;
+import org.sagebionetworks.bridge.upload.UnzipHandler;
+import org.sagebionetworks.bridge.upload.UploadValidationContext;
+
+/**
+ * <p>
+ * Bulk download utility. This downloads from s3 and decrypts (if it's decrypted) and writes the file to disk. This is
+ * basically a "do as much as you can" tool parallel to the BridgeUploadDownloader to quick check files, taking in a
+ * list of S3 files instead of a date range.
+ * </p>
+ * <p>
+ * Usage: play "run-main org.sagebionetworks.bridge.util.BulkDownloadUtil [S3 key1] [[S3 key2] [S3 key3] ...]"
+ * </p>
+ * <p>
+ * You'll also need to override the upload bucket, the CMS cert bucket, and CMS priv key bucket in your configs.
+ * </p>
+ */
+public class BulkDownloadUtil {
+    public static void main(String[] args) {
+        if (args.length == 0) {
+            System.out.println(
+                    "Usage: play \"run-main org.sagebionetworks.bridge.util.BulkDownloadUtil [S3 key1] [[S3 key2] [S3" +
+                            " key3] ...]\"");
+            System.exit(1);
+            return;
+        }
+        String[] s3KeyArr = args;
+
+        // Make tmp directory (if it doesn't exist). This has to be in the source root, since we're probably running
+        // this script in Vagrant.
+        File tmpDir = new File("tmp");
+        if (!tmpDir.exists()) {
+            tmpDir.mkdir();
+        }
+
+        System.out.println(String.format("Downloading %s files for S3 keys (%s)", s3KeyArr.length,
+                Joiner.on(", ").join(s3KeyArr)));
+
+        // spring beans
+        ApplicationContext springCtx = new ClassPathXmlApplicationContext("application-context.xml");
```

Comment:
You might want to call registerShutdownHook() to gracefully shut down the container and properly close the resources.